client/src/utils/filtering.ts

```diff
@@ -323,12 +344,39 @@ export default class Filtering<T> {
      * @returns {Boolean} True if the filter is set to the given value
      * */
     checkFilter<T>(filterText: string, filterName: string, filterValue: T): boolean {
-        const re = new RegExp(`${filterName}:(\\S+)`);
-        const reMatch = re.exec(filterText);
-        const testValue = reMatch ? reMatch[1] : this.defaultFilters[filterName];
+        const testValue = this.getFilterValue(filterText, filterName);
         return toLowerNoQuotes(testValue) === toLowerNoQuotes(filterValue);
     }
 
+    /** Get the value of a particular filter from filterText.
+     * @param {String} filterText Raw filter text string
+     * @param {String} filterName Filter key to check
+     * @returns {String | Boolean} The filterValue for the filter
+     * */
+    getFilterValue(filterText: string, filterName: string, alias = "eq"): string | boolean {
+        const op = getOperatorForAlias(alias);
+        const reInQuotes = `'([^']*[^\\s']*)'`;
+        const reNoQuotes = `(\\S+)`;
+        // Array of re groups, note: order matters as in quote filterVals must be checked first
+        const reGroups = [
+            `${filterName}${op}${reInQuotes}`,
+            `${filterName}${op}${reNoQuotes}`,
+            `${filterName}-${alias}:${reInQuotes}`,
+            `${filterName}-${alias}:${reNoQuotes}`,
+            `${filterName}_${alias}:${reInQuotes}`,
+            `${filterName}_${alias}:${reNoQuotes}`,
```

Comment:
These variations, can be combined into a single regex I think.

Reply:
Done! It is a single line regex now